lib/IAMClient.js

```diff
@@ -545,8 +545,9 @@ class IAMClient {
                 return callback(null, obj, res.statusCode, res.statusMessage);
             }
             // Load the error from errors(arsenal)
-            return callback(
-                errors[obj.ErrorResponse.Error.Code], null, res.statusCode);
+            const retErr = obj.ErrorResponse
```

Comment:
getObj() should never return `{}` if the http code is >= 400. It should either container object from parsed XML or object from parsed stringified arsenal error.
I did think about adding a fallback, just in case, but fallbacks are bad especially for the clients and we should fail hard at this point.